tests/Stream/SetTest.php

```diff
@@ -51,17 +51,251 @@ public function dataProviderForArray(): array
+                    ->toArray(),
+                [2, 3, 4],
+            ],
+            [
+                [
+                    [1, 2],
+                    [2, 3, 4],
+                    [2, 3, 4, 5, 6],
+                    3,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionNonStrictWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [2],
+            ],
+            [
+                [
+                    [1, 2],
+                    ['2', 3, 4],
+                    [2, 3, 4, 5, 6],
+                    1,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionNonStrictWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [1, '2', 3, 4, 5, 6],
+            ],
+            [
+                [
+                    [1, 2],
+                    ['2', 3, 4],
+                    [2, 3, 4, 5, 6],
+                    2,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionNonStrictWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [2, 3, 4],
+            ],
+            [
+                [
+                    [1, 2],
+                    ['2', 3, 4],
+                    [2, 3, 4, 5, 6],
+                    3,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionNonStrictWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [2],
+            ],
+            [
+                [
+                    [1, 2],
+                    [2, 3, 4],
+                    [2, 3, 4, 5, 6],
+                    1,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [1, 2, 3, 4, 5, 6],
+            ],
+            [
+                [
+                    [1, 2],
+                    [2, 3, 4],
+                    [2, 3, 4, 5, 6],
+                    2,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionWith(array_pop($iterables), ...$iterables)
+                    ->toArray(),
+                [2, 3, 4],
+            ],
+            [
+                [
+                    [1, 2],
+                    [2, 3, 4],
+                    [2, 3, 4, 5, 6],
+                    3,
+                ],
+                fn (array $iterables) => Stream::of(array_shift($iterables))
+                    ->partialIntersectionWith(array_pop($iterables), ...$iterables)
```

Comment:
Suggestion: These tests are too tricky. A user should be able to look at a test and immediately know what it does. Having unnecessary logic in the tests makes it hard to understand, and could hide bugs in the test code. How about just putting the `$minIntersectionCount` value in the stream method call directly to avoid the `array_pop` and mutating iterables to remove the non-iterable value.

Reply:
Fixed.